File: FRONT_END_CAR/src/app/Shared/models/rental-request.ts
export interface RentalRequest {
  rentalRequestId: string;
  customerId: string;
  carId: string;
  startDate: string;
  endDate: string;
  status: number;
  createDate: string;
  requestDate: string;
  updateDate: string | null;
}

export interface RentalRequest1 {
  customerId: string;
  carId: string;
  startDate: string;
  endDate: string;
}

File: BACK_END_CAR/CAR_RENTAL_API/Entities/RentalRequest.cs
using CAR_RENTAL_API.Enum;
using System.ComponentModel.DataAnnotations;

namespace CAR_RENTAL_API.Entities
{
    public class RentalRequest
    {
        [Key]
        public Guid RentalRequestId { get; set; } = Guid.NewGuid();

        [Required]
        public Guid CustomerId { get; set; }

        [Required]
        public Guid CarId { get; set; }

        [Required]
        public DateTime StartDate { get; set; }

        [Required]
        public DateTime EndDate { get; set; }

        [Required]
        public RentalRequestStatus? Status { get; set; }

        [Required]
        public DateTime RequestDate { get; set; } = DateTime.Now;


        public DateTime CreateDate { get; set; }

        public DateTime? UpdateDate { get; set; }
    }
}

